server/src/main/java/org/apache/druid/server/lookup/cache/LookupLoadingSpec.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.lookup.cache;
+
+import org.apache.druid.java.util.common.ISE;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class defines the spec for loading of lookups for a given task. It contains 2 fields:
+ * <ol>
+ *   <li>{@link LookupLoadingSpec#mode}: This mode defines whether lookups need to be
+ *   loaded for the given task, or not. It can take 3 values: </li>
+ *   <ul>
+ *    <li> ALL: Load all the lookups.</li>
+ *    <li> NONE: Load no lookups. </li>
+ *    <li> PARTIAL: Load only the lookups defined in lookupsToLoad </li>
+ *   </ul>
+ * <li>{@link LookupLoadingSpec#lookupsToLoad}: Defines the lookups to load when the lookupLoadingMode is set to PARTIAL.</li>
+ * </ol>
+ */
+public class LookupLoadingSpec
+{
+  public enum Mode
+  {
+    ALL, NONE, PARTIAL
+  }
+
+  private final Mode mode;
+  private final List<String> lookupsToLoad;
+
+  public static final LookupLoadingSpec ALL = new LookupLoadingSpec(Mode.ALL, null);
```

Comment:
Love the pojo. 
Nit: I would just make take in a list and change it to a immutable and then return an immutable list in the getter.

Reply:
Have made the change.

On second thoughts, do you see any downsides of moving to ImmutableSet instead of List? I think it makes sense since a given SqlNode can contain the same lookup name multiple times (I think?), so we wouldn't have to take care of the duplicates. Thoughts?

Reply:
I don't think we expect to have duplicates but it makes sense to have an ImmutableSet. We don't really care about the order of elements as they are just used for filtering.

Reply:
Have made the change to use ImmutableSet instead of ImmutableList.